generic/tests/fillup_disk.py

```diff
@@ -22,37 +59,53 @@ def run(test, params, env):
     login_timeout = int(params.get("login_timeout", 360))
     session = vm.wait_for_login(timeout=login_timeout)
     session2 = vm.wait_for_serial_login(timeout=login_timeout)
+    filled_partition = params.get("guest_testdir")
 
-    fillup_timeout = int(params.get("fillup_timeout"))
-    fillup_size = int(params.get("fillup_size"))
-    fill_dir = params.get("guest_testdir", "/tmp")
-    filled = False
-    number = 0
+    logging.debug('Check info of parttions: ')
+    parts_info = session.cmd_output('lsblk')
+    logging.debug(parts_info)
+    all_partitions = [str(mp.split()[-1]) for mp in parts_info.splitlines()
+                      if re.search(r'\d+\s+(lvm|part)\s+/', mp)]
 
     try:
-        error_context.context("Start filling the disk in %s" % fill_dir,
-                              logging.info)
-        cmd = params.get("fillup_cmd")
-        while not filled:
-            # As we want to test the backing file, so bypass the cache
-            tmp_cmd = cmd % (fill_dir, number, fillup_size)
-            logging.debug(tmp_cmd)
-            s, o = session.cmd_status_output(tmp_cmd, timeout=fillup_timeout)
-            if "No space left on device" in o:
-                logging.debug("Successfully filled up the disk")
-                filled = True
-            elif s != 0:
-                test.fail("Command dd failed to execute: %s" % o)
-            number += 1
+        if not filled_partition:
+            # fill up all the partitions exclude [SWAP].
+            for partition in all_partitions:
+                fill_partitions(test, params, session, partition)
```

Comment:
it seems that when you call `fill_partitions`, there is no situation that `deeply=True`

Reply:
The parameter deelp is actually work inside function fill_partitions,  however, if here call it with deelpy=True, fill_partitions will not work with costumer's params, will be force to called with 512b block size to run dd.